clang/test/Driver/sycl-offload-jit.cpp

```diff
@@ -1,9 +1,9 @@
 /// Perform several driver tests for SYCL offloading for JIT
 
 /// Check the phases graph with -fsycl. Use of -fsycl enables offload
-// RUN: %clang -ccc-print-phases --target=x86_64-unknown-linux-gnu -fsycl %s 2>&1 \
+// RUN: %clang --offload-new-driver -ccc-print-phases --target=x86_64-unknown-linux-gnu -fsycl %s 2>&1 \
```

Comment:
Can we create new substs in upstream lit.cfg.py so that we can add "--offload-new-driver" in downstream branches?
eg: add %clang_off and %clang_off_cl , they are equal to %clang and %clang_cl in llorg.
but we add --offload-new-driver in downstream branches?

Reply:
This may make some sense to do based on the differences in default behavior between intel/llvm and the community.  @sarnex is also running into this kinds of things on the OpenMP side.  Thoughts on this, Nick?

Reply:
The idea is to use the new subst eg: %clang_off for test that rely on new driver behaviors.

Reply:
yeah that seems fine to me, who is gonna implement it?

Reply:
Let us use this PR to fix the failures for now, implement the new subst in llorg later. Thanks.

Reply:
> Can we create new substs in upstream lit.cfg.py so that we can add "--offload-new-driver" in downstream branches?

Shouldn't downstream branches be aligned with the upstream regarding the offload driver model? I would bet that they will pushback and just say "use new offloading model by default, stop using old model!". 🤷

Reply:
We won't add old model upstream, just a dedicated subst in lit.cfg.py to be used in the affected tests.
We then add additional customizations downstream easily.

Reply:
I understand that, but what is the motivation for upstream to add dedicated subst for things that do not exist in upstream? How the upstream developers know which variable to use to not break the downstream?

Will it work if we add subst variables downstream only?

Reply:
We can't add subst downstream only -- that mean we have to update every %clang in test files, that doesn't really save us customizations.

Reply:
We can try add additional_flags downstream though.
```
llvm_config.use_clang(**additional_flags**=...)
```

Reply:
Although I'm skeptical about the community acceptance of this idea, it should not prevent you from trying... ;)

I would rather focus on switching DPC++ compiler to new offload model to avoid divergence with the community code.

Reply:
>rather focus on switching DPC++ compiler to new offload model to avoid divergence with the community code.

This is actually the same goal, to save the *switching* effort -- if we don't try to fix this in lit.cfg.py, we may end up need to maintain lots of driver tests during the switch for every PR we committed to llorg.